Controls/Mailman.c

```diff
@@ -4,19 +4,22 @@
  * The mailman will put mail in the mailbox.
  */
 
+
+/**
+ * @brief Puts a letter in the mailbox (declared in Tasks.h). The letter must be unique every time
+ * this function is called (suggestion: include a random number!). Print "Finished!" at the end.
+ */
+void depositLetter(void) {
+	OS_ERR err; // Make sure to check for errors and print the error code if not OS_ERR_NONE
+}
+
 /**
- * @brief Waits on the mailbox to be signaled. Once the mailbox is signaled, call depositLetter. Once the letter is deposited, signal the semaphore (effectively putting up the flag). Loop.
+ * @brief Calls depositLetter. Once the letter is deposited, signal the semaphore (effectively putting up the flag).
+ * Delay briefly (2-3 seconds) using OSTimeDlyHMSM(). Loop.
```

Comment:
Why does the time delay have to be in the task function rather than the helper?

Reply:
We can only post to the semaphore after the helper, so if the delay is in the helper, the mailbox will be signaled, and then the helper might be immediately called again, which would touch the value in the mailbox. Having the delay in the task (after the helper is called) ensures that the recipient can run and read what's in the mailbox.

Reply:
Additionally, should the time to read the letter be shorter than the time to deposit the letter? Otherwise, doesn't the semaphore get signaled more than once per read? Or does it even matter

Reply:
The overall reading time should be shorter than the overall depositing time. Otherwise, you'll eventually run into the situation where a letter will be deposited before the previous one is read, and information will be lost.

Reply:
Should we put that in a comment? Or explicitly say the amount of time they should delay on this task?

Reply:
Maybe have that in the comment. The exact times aren't super relevant other than their relative lengths.